src/common/dnd/drop.directive.ts

```diff
@@ -27,6 +29,7 @@ export class DropDirective implements OnInit, OnDestroy {
     this.listener.on('dragover', this.over);
     this.listener.on('dragleave', this.leave);
     this.listener.on('drop', this.drop);
+    this.canDrop = this.canDropFunction(this.model);
```

Comment:
i think this should be dynamic, depending on dragging stuff so it should be a function